satellite/console/consoleweb/server.go

```diff
@@ -260,58 +301,56 @@ func (server *Server) accountActivationHandler(w http.ResponseWriter, r *http.Re
 			zap.String("token", activationToken),
 			zap.Error(err))
 
-		server.serveError(w, r)
+		// TODO: when new error pages will be created - change http.StatusNotFound on appropriate one
+		server.serveError(w, r, http.StatusNotFound)
 		return
 	}
 
-	http.ServeFile(w, r, filepath.Join(server.config.StaticDir, "static", "activation", "success.html"))
+	if err = server.templates.activated.Execute(w, nil); err != nil {
+		server.serveError(w, r, http.StatusNotFound)
+	}
 }
 
 func (server *Server) passwordRecoveryHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	defer mon.Task()(&ctx)(nil)
+
 	recoveryToken := r.URL.Query().Get("token")
 	if len(recoveryToken) == 0 {
-		server.serveError(w, r)
+		server.serveError(w, r, http.StatusNotFound)
 		return
 	}
 
 	switch r.Method {
 	case http.MethodPost:
 		err := r.ParseForm()
 		if err != nil {
-			server.serveError(w, r)
+			server.serveError(w, r, http.StatusNotFound)
 			return
 		}
 
 		password := r.FormValue("password")
 		passwordRepeat := r.FormValue("passwordRepeat")
 		if strings.Compare(password, passwordRepeat) != 0 {
-			server.serveError(w, r)
+			server.serveError(w, r, http.StatusNotFound)
 			return
 		}
 
 		err = server.service.ResetPassword(ctx, recoveryToken, password)
 		if err != nil {
-			server.serveError(w, r)
+			server.serveError(w, r, http.StatusNotFound)
 			return
 		}
 
-		http.ServeFile(w, r, filepath.Join(server.config.StaticDir, "static", "resetPassword", "success.html"))
-	case http.MethodGet:
-		t, err := template.ParseFiles(filepath.Join(server.config.StaticDir, "static", "resetPassword", "resetPassword.html"))
-		if err != nil {
-			server.serveError(w, r)
-			return
+		if err := server.templates.success.Execute(w, nil); err != nil {
+			server.serveError(w, r, http.StatusNotFound)
```

Comment:
For safety return immediately after `serveError`, because often people add some code at the end of the function, and if you don't return from the error handling it could end up executing that code.